fluid/fluid.cpp

```diff
@@ -38,6 +38,11 @@ namespace FluidS {
 
 bool Fluid::initialized = false;
 
+/* better than a macro to determine inappropriate values for notes*/
+bool validNote(const int input) {
+      return (input < 255 && input > -1);
```

Comment:
shouldn't it be `input < 256` or `input <= 255`?

Reply:
Good question. At another place in this file this is claimed to "usually be the TPC" (Tonal Pitch Class), but that  in turn has a range of -2 (`TPC_INVALID`) to 33 (`TPC_B_SS`), so that doesn't seem to be the case.

Reply:
It it were pitches though, the limit would rather be 0-127

Reply:
However, as far as I can tell this `validNote()` is call only from `Fluid::setFromKeyPortamento()` and that in turn is never called!?!

Reply:
This code was copied from fluid synth 2, which isn't quite possible to completely incorporate into MuseScore. Jojo could actually be right, I just wasn't entirely too sure what's necessary for portamento and what's not, or what might be necessary in the future, and I wanted to add as much of fluid synth 2's extra features as possible. It's specifically to do with line 219 of this file:

https://github.com/FluidSynth/fluidsynth/blob/master/src/synth/fluid_chan.h#L219

If the file has changed, look for a macro like `#define fluid_channel_is_valid_note(n)    (n != INVALID_NOTE)` I'm not a fan of macros because I think they can occasionally be unpredictable and otherwise some compilers may or may not optimise them well or add unnecessary symbols.

Long story short, 255 is indeed an invalid value as is any value that's negative.

Reply:
* So `const int INVALID_NOTE = 255;` rather than a magic number?
Instead with have this though:

      enum { INVALID_NOTE = -1};

Confusing? Yes...

* That comment about TPC seems missleading maybe it means PTC, PortamenTo Controler? Or is it really TPC?